src/backend/access/external/hd_work_mgr.c

```diff
@@ -768,6 +769,16 @@ make_allocation_output_string(List *segment_fragments)
 	initStringInfo(&segwork);
 	appendStringInfoString(&segwork, SEGWORK_PREFIX);
 	
+	if (enable_secure_filesystem)
```

Comment:
if we append the dfs_address through the external framework, why do we need to append it here as well? wouldn't it work for both readable and writable tables?